RELEASES.md

```diff
@@ -1,3 +1,130 @@
+Version 1.54.0 (2021-06-29)
+============================
+
+Language
+-----------------------
+
+- [You can now use macros for values in key-value attributes.][83366]
+  While a seemingly minor addition on its own, this enables a lot of
+  powerful functionality when combined correctly. Most notably you can
+  now include external documentation in your crate by writing the following.
+  ```rust
+  #![doc = include_str!("README.md")]
+  ```
+
+- [You can now cast between unsized slice types (and types which contain
+  unsized slices) in `const fn`.][85078]
+- [TODO: stabilize member constraints][84701]
+
+Compiler
+-----------------------
+
+- [Rustc will now search for custom JSON targets in
+  `/lib/rustlib/<target-triple>/target.json` where `/` is the "sysroot"
+  directory.][83800] You can find your sysroot directory by running
+  `rustc --print sysroot`.
+- [Added `wasm` as a `target_family` for WebAssembly platforms.][84072]
+- [You can now use `#[target_feature]` on safe functions when targeting
+  WebAssembly platforms.][84988]
+- [Improved debugger output for enums on Windows MSVC platorms.][85292]
+- [Added tier 3\* support for `bpfel-unknown-none`
+   and `bpfeb-unknown-none`.][79608]
+- [Added tier 3 support for `powerpc64le-unknown-freebsd`.][83572]
+
+Libraries
+-----------------------
+
+- [You can now explicitly import the prelude of different editions
+  through `std::prelude`. (e.g. `use std::prelude::rust_2021::*;`)][86294]
+- [`panic::panic_any` will now `#[track_caller]`.][85745]
+- [Added `OutOfMemory` as a variant of `io::ErrorKind`.][84744]
+- [ `proc_macro::Literal` now implements `FromStr`.][84717]
+- [TODO: Bump stdarch submodule][83278]
```

Comment:
Not sure what to write for this one, there's a lot of changes in one commit. Maybe we should be tracking `stdarch`  in relnotes generation to more easily include the specific commits added to `stdarch`.

Reply:
I can't speak to most of the changes, but this one:
> Intrinsics that previously used #[rustc_args_required_const] now use const generics. See Use const generics for stdarch intrinsics #83167 for more details.

is important a) because it greatly decreases the bootstrapping time for `core` (by about 50%!) and b) because it comes with some backwards-compatibility edge cases. It also unfortunately regresses binary sizes a bit because functions are monomorphized for each constant.

cc @Amanieu for the other changes

Reply:
There is no binary size regression: these functions were always being inlined anyways.

I'd summarize it as:
> The implementations of vendor intrinsics in `core::arch` have been significantly refactored. The main user-visible changes are a 50% reduction in the size of libcore.rlib and stricter validation of constant operands passed to intrinsics. The latter is technically a breaking change, but allows Rust to more closely match the C vendor intrinsics API.